components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1168,76 +1168,51 @@ end subroutine dropmixnuc
    ! below layer k  (k,k+1 interface)
-   real(r8), intent(in) :: ekkm(pver) ! zn*zs*density*diffusivity (kg/m3 m2/s) at interface
+   real(r8), intent(in) :: ekkm(pver) ! [/s] zn*zs*density*diffusivity (kg/m3 m2/s) at interface
    ! above layer k  (k,k+1 interface)
-   real(r8), intent(in) :: overlapp(pver) ! cloud overlap below
-   real(r8), intent(in) :: overlapm(pver) ! cloud overlap above
-   real(r8), intent(in) :: surfrate ! surface exchange rate (/s)
-   real(r8), intent(in) :: flxconv ! convergence of flux from surface
-   real(r8), intent(in) :: dt ! time step (s)
+   real(r8), intent(in) :: overlapp(pver) ! [fraction] cloud overlap below
+   real(r8), intent(in) :: overlapm(pver) ! [fraction] cloud overlap above
+   real(r8), intent(in) :: dt ! [s] time step 
    logical, intent(in) :: is_unact ! true if this is an unactivated species
-   real(r8), intent(in),optional :: qactold(pver)
-   ! mixing ratio of ACTIVATED species from previous step
+   real(r8), intent(in),optional :: qactold(pver) 
+   ! [# or kg / kg] number / mass mixing ratio of ACTIVATED species from previous step
    ! *** this should only be present
    !     if the current species is unactivated number/sfc/mass
 
    integer k,kp1,km1
 
-   if ( is_unact ) then
       !     the qactold*(1-overlap) terms are resuspension of activated material
       do k=top_lev,pver
          kp1=min(k+1,pver)
          km1=max(k-1,top_lev)
-         q(k) = qold(k) + dt*( - src(k) + ekkp(k)*(qold(kp1) - qold(k) +       &
-            qactold(kp1)*(1.0_r8-overlapp(k)))               &
-            + ekkm(k)*(qold(km1) - qold(k) +     &
-            qactold(km1)*(1.0_r8-overlapm(k))) )
-         !        force to non-negative
-         !        if(q(k)<-1.e-30)then
-         !           write(iulog,*)'q=',q(k),' in explmix'
-         q(k)=max(q(k),0._r8)
-         !        endif
-      end do
 
-      !     diffusion loss at base of lowest layer
-      q(pver)=q(pver)-surfrate*qold(pver)*dt+flxconv*dt
-      !        force to non-negative
-      !        if(q(pver)<-1.e-30)then
-      !           write(iulog,*)'q=',q(pver),' in explmix'
-      q(pver)=max(q(pver),0._r8)
-      !        endif
-   else
-      do k=top_lev,pver
-         kp1=min(k+1,pver)
-         km1=max(k-1,top_lev)
-         q(k) = qold(k) + dt*(src(k) + ekkp(k)*(overlapp(k)*qold(kp1)-qold(k)) +      &
-            ekkm(k)*(overlapm(k)*qold(km1)-qold(k)) )
+         if ( is_unact ) then
+            q(k) = qold(k) + dt*( - src(k) + ekkp(k)*(qold(kp1) - qold(k) +       &
+               qactold(kp1)*(1.0_r8-overlapp(k)))               &
+               + ekkm(k)*(qold(km1) - qold(k) +     &
+               qactold(km1)*(1.0_r8-overlapm(k))) )
+         else
+            q(k) = qold(k) + dt*(src(k) + ekkp(k)*(overlapp(k)*qold(kp1)-qold(k)) +      &
+               ekkm(k)*(overlapm(k)*qold(km1)-qold(k)) )
+         endif
          !        force to non-negative
          !        if(q(k)<-1.e-30)then
```

Comment:
get rid of this dead code and the `endif` as well.

Reply:
OK, done in the latest commit.